src/script/lua_api/l_server.cpp

```diff
@@ -433,6 +434,29 @@ int ModApiServer::l_get_worldpath(lua_State *L)
 	return 1;
 }
 
+// get_mod_data_path()
+int ModApiServer::l_get_mod_data_path(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	lua_rawgeti(L, LUA_REGISTRYINDEX, CUSTOM_RIDX_CURRENT_MOD_NAME);
+	if (!lua_isstring(L, -1)) {
+		return 0;
+	}
+
+	std::string modname = readParam<std::string>(L, -1);
+
+	const Server *srv = getServer(L);
+	std::string path = srv->getModDataPath() + DIR_DELIM + modname;
+	if (!fs::CreateAllDirs(path)) {
+		lua_pushstring(L, "Failed to create dir");
+		lua_error(L);
```

Comment:
```suggestion
		luaL_error(L, "Failed to create dir");
```

Reply:
better yet: `throw LuaError("Failed to create dir");`